test/unit/local/atom/filter_ignored.js

```diff
@@ -29,142 +33,186 @@ onPlatforms(['linux', 'win32'], () => {
-        channel.push(batch)
+        channel.push(
+          _.cloneDeep([
+            notIgnoredScanEvent,
+            ignoredScanEvent,
+            notIgnoredCreatedEvent,
+            ignoredCreatedEvent,
+            notIgnoredDeletedEvent,
+            ignoredDeletedEvent,
+            ignoredRenamedSrcEvent,
+            ignoredRenamedDstEvent,
+            notIgnoredRenamedEvent,
+            INITIAL_SCAN_DONE
+          ])
+        )
       })
 
       it('keeps only the relevant events in order', async () => {
         const filteredChannel = filterIgnored.loop(channel, { ignore })
 
+        const renamedToCreatedEvent = {
+          ...ignoredRenamedSrcEvent,
+          action: 'created',
+          [filterIgnored.STEP_NAME]: {
+            movedFromIgnoredPath: ignoredRenamedSrcEvent.oldPath
+          }
+        }
+        delete renamedToCreatedEvent.oldPath
+        const renamedToDeletedEvent = {
+          ...ignoredRenamedDstEvent,
+          action: 'deleted',
+          // $FlowFixMe ignoredRenamedDstEvent does have an oldPath attribute
+          _id: metadata.id(ignoredRenamedDstEvent.oldPath),
+          path: ignoredRenamedDstEvent.oldPath,
+          [filterIgnored.STEP_NAME]: {
+            movedToIgnoredPath: ignoredRenamedDstEvent.path
+          }
+        }
+        delete renamedToDeletedEvent.oldPath
         const relevantEvents = await filteredChannel.pop()
-        should(relevantEvents).deepEqual(notIgnoredEvents)
+        should(relevantEvents).deepEqual([
+          notIgnoredScanEvent,
+          notIgnoredCreatedEvent,
+          notIgnoredDeletedEvent,
+          renamedToCreatedEvent,
+          renamedToDeletedEvent,
+          notIgnoredRenamedEvent,
+          INITIAL_SCAN_DONE
+        ])
       })
     })
 
     context('with multiple batches of file & folder events', () => {
-      const ignoredEvents = [
-        builders
-          .event()
-          .path('ignored1.bck')
-          .kind('file')
-          .build(),
-        builders
-          .event()
-          .path('ignored2.bck')
-          .kind('file')
-          .build()
-      ]
-      const notIgnoredEvents = [
-        builders
-          .event()
-          .path('notIgnored1.txt')
-          .kind('file')
-          .build(),
-        builders
-          .event()
-          .path('notIgnored2.txt')
-          .kind('file')
-          .build()
-      ]
       beforeEach(() => {
-        channel.push([ignoredEvents[0], notIgnoredEvents[0]])
-        channel.push([ignoredEvents[1], notIgnoredEvents[1]])
+        channel.push(_.cloneDeep([notIgnoredScanEvent, ignoredScanEvent]))
+        channel.push(_.cloneDeep([notIgnoredCreatedEvent, ignoredCreatedEvent]))
+        channel.push(_.cloneDeep([notIgnoredDeletedEvent, ignoredDeletedEvent]))
+        channel.push(
+          _.cloneDeep([ignoredRenamedSrcEvent, ignoredRenamedDstEvent])
+        )
+        channel.push(_.cloneDeep([notIgnoredRenamedEvent, INITIAL_SCAN_DONE]))
```

Comment:
Do we really need to test all those cases again when the *filter_ignore* is not aggregating batches in any way?

Reply:
I just wanted to make sure we have tests for both 1 batch and multi-batches. Besides it was already done that way.
But it might be overkill indeed.

Reply:
It's just we know some steps can have different behaviors depending on the batches they're given and event if this one does not __at the moment__ I'd feel safer it this was tested.